### PR TITLE
feat: Modify New Page to Select New Template

### LINKED ISSUE #590: Modify New Page to Select New Template
## Description
 
 AS A Content Editor
 I WANT to be able to select a new template from Edit mode
 SO THAT I can quickly create a new page that follows the sections governance. 

 - Wireframes: https://jhdfdc.axshare.com/#g=1&p=generic_ui&id=sxj4x9
## Acceptance Criteria
AC1: (EDITED on 12/7 to not move it to a submenu with one icon.
~~In the admin tool bar, when selected the Page button opens a sub panel for "Add New" with the "Blank" icon and label~~

AC2. 
~~When the user selects the "Blank" button the "Add a Blank Page" form appears~~

AC 1 & 2 When the user selects "Page" button from admin tool bar then form panel in AC3 is opened.

AC3. 

The AC2 form panel window includes:
 - Title "Add a Blank Page"
 - Template. This field is not editable and pulls in 
 -- the child page template* if it exists else
 -- it uses the current page template name if it is using the template, and if does not exist  
 -- proceeds up the parents to find a template in use and uses that, and if does not exist
 -- set to "default" 
 - Page Path. This field displays the parent path from the current page and the user can enter in a page URL
 - There is an Edit link that makes the full page path editable and the user can modify the entire string and allow for the "/"
  - If a parent page, in the path does not exist the folder structure will be created and the parent page will be rendered as a 404

AC4: 

The AC2/AC3 Panel window uses the expected behavior for the check and close functionality. 

AC5:
 The validation works the same as what is currently being used for the Add New Page capability. 
 If the User adds a URL name that already exists they will get an error notice that the page already exists.

AC6. 
 The Locked editable field is a reusable component; this includes the "Edit" link/button and the text field functionality

### PR DESCRIPTION
**Pull request by beliayeu**
## Changes
* added a sub panel for page button in admin menu
* added a "Blank" button to the page sub panel
* made add new page form according to AC3 from #590 
* made the field for entering page url reusable
* updated cypress tests with new page form selectors

## Related Issues

Closes #590

### CODE REVIEW

File: packages/gatsby-theme-bodiless/src/dist/NewPageURLField.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ */
+
+import React, { useCallback } from 'react';
+import { useMenuOptionUI, useNode } from '@bodiless/core';
+import { useField } from 'informed';
+import path from 'path';
+
+const BASE_PATH_FIELD_NAME = 'basePath';
+const PAGE_URL_FIELD_NAME = 'pagePath';
+const INPUT_FIELD_DEFAULT_CLASSES = 'bl-text-grey-900 bg-grey-100 bl-text-xs bl-min-w-xl-grid-1 bl-my-grid-2 bl-p-grid-1';
```
[the remark this answers is not in the document]
Comment (beliayeu) [reply packages/gatsby-theme-bodiless/src/dist/NewPageURLField.tsx]: nice catch, fixed in d94e9e93e2723f80874b9fbf8567cafa73aa1e51

**Review by VancheeZze — APPROVED** [review]

**Comment by TatsianaMiraniuk** [comment]
@beliayeu ,

please see **issue_1:**

**Steps to reproduce:**
1. open Edit env, any page, expect home page, e.g. /products/
2. switch to Edit mode
3. in Admin Menu click '+ Page'
4. click 'Blank'
5. in Add a Blank Page form, click Edit below Page path
**Actual Results:**
'No special characters or spaces allowed' message shows
Screen Shot 2020-11-25 at 3 13 17 PM


**Expected Results:**
no message is expected. 

NOTE:
the message disappears once a user clicks in the field

**Comment by TatsianaMiraniuk** [comment]
@beliayeu 

please see **issue_2:**

'Add New' is missed.
Screen Shot 2020-11-25 at 3 28 42 PM

**Comment by TatsianaMiraniuk** [comment]
@beliayeu ,

please see **issue_3**:
the whole row of 'Edit' link is clickable. Can we make clickable only the 'Edit' label?
Screen Shot 2020-11-25 at 4 36 09 PM

**Comment by TatsianaMiraniuk** [comment]
@beliayeu ,

please see **issue_4:**

**Steps to reproduce:**
1. open 'Add a Blank Page' form in any page
2. click inside the page path field, don't enter anything
3. click outside the field, but still in the Form to move out the focus from the field  => 'Field can not be empty' message shows.
4. click at the right of the message as shown in a screenshot below
Screen Shot 2020-11-25 at 4 41 17 PM

**Actual Results:**
The form closes

**Expected Results:**
the form is not supposed to be closed if a user click anywhere inside the form expect 'x' icon

**Comment by TatsianaMiraniuk** [comment]
@beliayeu ,


please see **issue_5:**
'my-page' text placeholder is missing:
Screen Shot 2020-11-30 at 9 54 29 AM

**Comment by TatsianaMiraniuk** [comment]
@beliayeu ,

please see **issue_6:**

**Steps to reproduce:**
1. open Edit env, any existing page, expect Home, e.g. /accordion/
2. switch to Edit mode
3. in Admin Menu click '+ Page'
4. click 'Blank'
5. in Add a Blank Page form, click Edit below Page path
until issue_1 is fixed, do the steps 6 and 7
6. click inside the field
7. click outside the field, but still in on the form (error message should disappear)
8. click 'v'

**Actual Results:**
page is being created. operation complete with a link to the current page.
index.json file is created in the page folder

**Expected Results:**
the error message 'Error: page  already exists' should be shown

**Comment by beliayeu** [comment]
@TatsianaMiraniuk,
issue_1, issue_3, issue_4, issue_5, issue_6 fixed
issue_2 moved from this ticket scope

**Review by VancheeZze — APPROVED** [review]

**Comment by beliayeu** [comment]
> @beliayeu , could you please recheck point 1? the placeholder is missed:

@TatsianaMiraniuk, nice catch
updated in 598aaed4e1dda1ff677bd1a95acdf13917ce510f
note, the same placeholder is in both modes ('Page Path' and 'URL')
@kbray0009, please advice if you want to have a different placeholder in 'URL' mode

**Review by VancheeZze — APPROVED** [review]

**Review by VancheeZze — APPROVED** [review]

**Review by VancheeZze — DISMISSED** [review]

**Review by VancheeZze — DISMISSED** [review]

**Review by VancheeZze — APPROVED** [review]

**Review by TatsianaMiraniuk — APPROVED** [review]

**Pull request merged**